utils/src/registerCommand.ts

```diff
@@ -78,15 +78,20 @@ async function setTelemetryProperties(context: types.IActionContext, args: unkno
 
     // handles items from the resource groups extension
     if (isResourceGroupsItem(firstArg)) {
-        context.telemetry.properties.resourceId = (firstArg as { resource: Resource })?.resource?.id;
+        const resourceId = (firstArg as { resource: Resource })?.resource?.id;
+        if (resourceId) {
+            context.telemetry.properties.resourceId = new TelemetryTrustedValue(resourceId);
+        }
         context.telemetry.properties.subscriptionId = (firstArg as { resource: Resource })?.resource?.subscription?.subscriptionId;
     }
 
     // handles items from v1 extensions
     for (const arg of args) {
         if (arg instanceof AzExtTreeItem) {
             try {
-                context.telemetry.properties.resourceId = arg.id;
```

Comment:
Can we change the logic here to only attach subscription ID and resource ID if subscription is _defined_? The Docker extension shouldn't be attaching these IDs but it is. It does not have subscriptions associated with `AzExtTreeItem`s.